internal/cloud/realm/auth_test.go

```diff
@@ -78,16 +79,58 @@ func TestRealmAuthRefresh(t *testing.T) {
 		assert.Equal(t, nil, err)
 
 		// invalidate the session's tokens
-		session.AccessToken = ""
 		session.RefreshToken = session.AccessToken
+		session.AccessToken = ""
 
 		profile := mock.NewProfileWithSession(t, session)
 
 		client = realm.NewAuthClient(profile.RealmBaseURL(), profile)
 		_, err = client.AuthProfile()
 		assert.Equal(t, realm.ErrInvalidSession{}, err)
 	})
-	// TODO: REALMC-7719 add test for expired credentials and test for ensuring profile cleared on invalid session
+
+	t.Run("with an expired access token", func(t *testing.T) {
+		u.SkipUnlessExpiredAccessTokenPresent(t)
+		t.Run("should use the refresh token to generate a new access token", func(t *testing.T) {
+			client := realm.NewClient(u.RealmServerURL())
+			u.SkipUnlessRealmServerRunning(t)
+			session, err := client.Authenticate(u.CloudUsername(), u.CloudAPIKey())
+			assert.Equal(t, nil, err)
+
+			session.AccessToken = u.ExpiredAccessToken() // Set the access token to an expired token
+
+			profile, teardown := mock.NewProfileFromTmpDir(t, "auth_refresh_test")
+			defer teardown()
+			profile.SetRealmBaseURL(u.RealmServerURL())
+			profile.SetSession(auth.Session{session.AccessToken, session.RefreshToken})
+
+			client = realm.NewAuthClient(profile.RealmBaseURL(), profile)
+			_, err = client.AuthProfile()
+			assert.Nil(t, err)
+		})
+		t.Run("should error if both the access and refresh tokens are expired", func(t *testing.T) {
+			client := realm.NewClient(u.RealmServerURL())
+			u.SkipUnlessRealmServerRunning(t)
+			session, err := client.Authenticate(u.CloudUsername(), u.CloudAPIKey())
+			assert.Equal(t, nil, err)
+
+			session.AccessToken = u.ExpiredAccessToken()  // Set the access token to an expired token
+			session.RefreshToken = u.ExpiredAccessToken() // Set the refresh token to an random expired token
```

Comment:
i think again, just pass the `u.ExpiredAccessToken()` values directly to the `SetSession` call.

you can add a comment there saying:

```go
// set the access and refresh token values to expired tokens
profile.SetSession(auth.Session{u.ExpiredAccessToken(), u.ExpiredAccessToken()})
```